docs/architecture/adr-010-posts-module.md

```diff
@@ -0,0 +1,345 @@
+  required uint64 post_id = 2;
+  
+  // Attachment to be added to the post 
+  required Attachment attachment = 3;
+
+  // Author of the post
+  required string author = 4;
+  
+  message Attachment {
+    oneof sum {
+      Poll poll = 1;
+      Media media = 2;
+    }
+  }
+}
+
+// MsgAddPostAttachmentResponse defines the Msg/AddPostAttachment response type.
+message MsgAddPostAttachmentResponse {
+  // New id of the uploaded attachment 
+  required uint32 attachment_id = 1;
+}
+
+// MsgRemovePostAttachment represents the message to be used when 
+// removing an attachment from a post
+message MsgRemovePostAttachment {
+  // Id of the subspace containing the post
+  required uint64 subspace_id = 1;
+
+  // Id of the post from which to remove the attachment
+  required uint64 post_id = 2;
+  
+  // Id of the attachment to be removed
+  required uint32 attachment_id = 3;
+
+  // User that is removing the attachment
+  required string signer = 4;
+}
+
+// MsgRemovePostAttachmentResponse defines the 
+// Msg/RemovePostAttachment response type.
+message MsgRemovePostAttachmentResponse {}
+
+// MsgDeletePost represents the message used when deleting a post.
+message MsgDeletePost {
+  // Id of the subspace containing the post
+  required uint64 subspace_id = 1;
+
+  // Id of the post to be deleted
+  required uint64 post_id = 2;
+  
+  // User that is deleting the post
+  required string signer = 3;
+}
+
+// MsgDeletePostResponse represents the Msg/DeletePost response type
+message MsgDeletePostResponse {}
+
+// MsgAnswerPoll represents the message used to answer a poll
+message MsgAnswerPoll {
+  // Id of the subspace containing the post
+  required uint64 subspace_id = 1;
+
+  // Id of the post that contains the poll to be answered
+  required uint64 post_id = 2;
+  
+  // Id of the poll to be answered 
+  required uint32 poll_id = 3;
+  
+  // Index of the answer inside the ProvidedAnswers array
+  required uint32 answer_index = 4;
+}
+
+// MsgAnswerPollResponse represents the MSg/AnswerPoll response type
+message MsgAnswerPollResponse {}
+```
+
+## Consequences
+
+### Backwards Compatibility
+
+The changes described inside this ADR are **not** backward compatible. To solve this, we will rely on the `x/upgrade` module in order to properly add these new features inside a running chain. If necessary, to make sure no extra operation is performed, we should make sure that `fromVm[poststypes.ModuleName]` is set to `1` before running the migrations, so that the `InitGenesis` method does not get called.
+
+### Positive
+
+- Allows users to create content inside an application
+
+### Negative
+
+- Not known
+
+### Neutral
+
+- Required the `x/subspaces` to implement two new permissions: 
+   - `PermissionManageContent` to allow moderators to remove post attachment and posts from a subspace; 
+   - `PermissionCreteContent` to allow users to create a content inside a subspace.
+
+## Further Discussions
+
+## References
+
```

Comment:
I consider them base tests, that's why I didn't write them all: I take them for granted as they are base tests that will make sure the module works properly. We've never listed them all (as it would just take a lot of time and not be truly beneficial) anyway